Update fish support

gazorby opened the pull request:
## WHAT
- Solve issues during installation on fish shell (#107, #108, #123, #124, #132 )
- Focus on [fisher](https://github.com/jorgebucaran/fisher) package manager, drop omf support
- Add `alt+f` keybinding to trigger enhancd when writing commands and paste the selected item (fish)
- Add git root inside a git repo (#110, #111) (fish & bash)

## WHY
Solve annoying fish issues (which i introduced with #99) & bring more fish love :heart: :fish:

## TODO

~~Currently `ENHANCD_ROOT` still reference my fork. If this PR is considered and approved, it should be modified to point the correct plugin path (`b4b4r07` instead of `gazorby`)~~

- [x] update `ENHANCD_ROOT`

jorgebucaran commented:
@gazorby Good job! Further changes are required to work with the latest Fisher, see https://github.com/jorgebucaran/fisher/issues/606. Other than your changes, we also need to place `lib` inside `functions` and [replace this line](https://github.com/b4b4r07/enhancd/blob/f0f894029d12eecdc36c212fa3ad14f55468d1b7/conf.d/enhancd.fish#L15) with the correct location, i.e. `$fisher_path/functions`.

gazorby commented:
@jorgebucaran done ;)

setting `fisher_path` to `~/.config/fish` seems to work properly

jorgebucaran commented:
Hey, @gazorby, `fisher_path` should not be set by a plugin, as that would completely break the user's installation. 

To use the latest Fisher, we really just need to move `lib` to a sensible location under `functions`, e.g. `functions/enhancd/lib` and update `ENHANCD_ROOT` appropriately.

jorgebucaran reviewed (changes requested):
@gazorby Good work! 

- We should move `functions/lib` to `functions/enhancd/lib` (for safety).
- Update `$ENHANCD_ROOT` appropriately.
  - You may use `$fisher_path/functions/enhancd`.

gazorby commented:
> > setting fisher_path to ~/.config/fish seems to work properly
> 
> Hey, @gazorby, `fisher_path` should not be set by a plugin, as that would completely break the user's installation. I didn't see where you were setting it in this PR, though, so I'm just addressing your comment.
> 
> Now, to use the latest Fisher, we really just need to move `lib` to a sensible location under `functions`, e.g. `functions/enhancd/lib` and update `ENHANCD_ROOT` appropriately.

Sorry for being misunderstood, I know that `fisher_path` should be set by the user, that was just my local settings, I didn't change `fisher_path` in this PR.
I mentioned it as I first tried to set it to `~/.config/fisher` but it didn't work because I didn't know that I had to manually source conf.d snippets (so `cd` is aliased to `enhancd`).

jorgebucaran commented on functions/_enhancd_filepath_abs.fish:
```diff
@@ -3,13 +3,13 @@ function _enhancd_filepath_abs
     set -l dir "$argv"
 
-    if [ -z "$dir" ] || [ -p /dev/stdin ]
+    if test -z "$dir"; or test -p /dev/stdin
```
@gazorby We are already using fish `3.x` new features in some places, so might as well use `||`, rather than `; or` here.

jorgebucaran commented on conf.d/enhancd.fish:
```diff
@@ -1,64 +1,70 @@
-# * $dependencies  package dependencies
+function __enhancd_install --on-event enhancd_install
+    # set variables
+    set -Ux ENHANCD_FILTER
```
@gazorby Why the universal scope?

gazorby replied:
@jorgebucaran because `__enhancd_install` is called only on plugin install and `ENHANCD_FILTER` is accessed each time `enhancd` is used.

jorgebucaran replied:
@gazorby It's one way, but not the only way. We could also set them outside the event handler since `conf.d` snippets are loaded on startup.

gazorby replied:
@jorgebucaran That was how it was done before fisher 4.0, I'm just trying to leverage fisher event handlers from your latest release. I feel that they are appropriate for this kind of situation, It avoids bloating shell startup and removes the need to check for variables existence (`set -q var; or set -gx var value`), making the snippet a bit more readable.

jorgebucaran commented on functions/_enhancd_entry_git_root.fish:
```diff
@@ -0,0 +1,8 @@
+function _enhancd_entry_git_root
+    if git rev-parse --is-inside-work-tree 2> /dev/null
+       and test (git rev-parse --show-toplevel) != (pwd)
```
@gazorby I have nothing against `and`, but if we're going to be using `&&` in some places, I'd prefer to use `&&` everywhere.

gazorby replied:
@jorgebucaran I agree with the syntax consistency, but as `&&` and `||` were introduced for compatibility with POSIX shells and we have separate sources for bash/zsh, I'd prefer to use the plain `and` / `or` fish syntax.

jorgebucaran reviewed (approved):
LGTM

jorgebucaran commented:
@b4b4r07 Can you review this PR? 👋

jorgebucaran commented:
@gazorby Looks good me. ✅ 

@b4b4r07 What else do you need to merge here?

jorgebucaran commented:
@gazorby What else do you we need to merge here? Have you checked that this doesn't break with a Bash or Zsh installation?

gazorby commented:
Indeed, bash/zsh was broken since I forgot to update references after moving the `lib` folder to the fish directory. @kshenoy's PR fixed it.

I just checked using zsh with zplug and didn't found any issue.

gazorby reviewed (commented):
LGTM

jorgebucaran commented:
Any idea why the CI is failing?

gazorby commented:
Fixed test CI.

I think that triage is failing because secrets aren't accessible on pull request from forks.
